custom_model_runner/datarobot_drum/drum/drum.py

```diff
@@ -1164,3 +1172,71 @@ def create_custom_inference_model_folder(code_dir, output_dir):
         fp.write(readme)
     if files_in_output & copied_files:
         print("Files were overwritten: {}".format(files_in_output & copied_files))
+
+
+def get_default_parameter_values(model_metadata: Dict) -> Dict[str, Union[int, float, str]]:
+    """Retrieve default parameter values from the hyperparameter section of model-metadata.yaml.
+    When `default` is provided, return the default value.
+    When `default` is not provided:
+        - Return `min` value if it is a numeric parameter.
+        - Return an emtpy string if it is a string parameter.
+        - Return the first allowed value if it is the select parameter.
+        - Return the default value of the first component parameter (sorted by parameter type) if it is a multi
+          parameter.
+    """
+
+    def _get_default_numeric_param_value(param_config: Dict) -> Union[int, float]:
+        """Get default value of numeric parameter."""
+        param_default_value = param_config.get("default")
+        if param_default_value is not None:
+            return param_default_value
+        else:
+            return param_config["min"]
+
+    def _get_default_string_param_value(param_config: Dict) -> str:
+        """Get default value of string parameter."""
+        param_default_value = param_config.get("default")
+        if param_default_value is not None:
+            return param_default_value
+        else:
+            return ""
+
+    def _get_default_select_param_value(param_config: Dict) -> str:
+        """Get default value of select parameter."""
+        param_default_value = param_config.get("default")
+        if param_default_value is not None:
+            return param_default_value
+        else:
+            return param_config["values"][0]
+
+    def _get_default_multi_param_value(param_config: Dict) -> str:
+        """Get default value of multi parameter."""
+        param_default_value = param_config.get("default")
+        if param_default_value is not None:
+            return param_default_value
+        else:
+            param_values = param_config["values"]
+            if param_values:
+                first_component_param_type = sorted(param_values.keys())[0]
+                first_component_param = param_values[first_component_param_type]
+                if first_component_param_type in {"int", "float"}:
+                    return _get_default_numeric_param_value(first_component_param)
+                elif first_component_param_type == "select":
+                    return _get_default_select_param_value(first_component_param)
+
+    hyper_param_config = model_metadata.get(ModelMetadataKeys.HYPERPARAMETERS, [])
+    default_params = {}
+    for param_config in hyper_param_config:
+        param_name = param_config["name"]
+        param_type = param_config["type"]
+        if param_type in {"int", "float"}:
```

Comment:
could you define global vars for "int" "float" "select" "string" and "multi" and refactor to use them? Probably would prevent typos.

Reply:
updated
https://github.com/datarobot/datarobot-user-models/blob/3f5e21b85c9a075401eb3cf13e13975c15648b97/custom_model_runner/datarobot_drum/drum/drum.py#L1212-L1217